apkInspector/extract.py

```diff
@@ -42,9 +42,13 @@ def extract_file_based_on_header_info(apk_file, local_header_info, central_direc
         # -15 for windows size due to raw stream with no header or trailer
         extracted_data = zlib.decompress(compressed_data, -15)
         indicator = 'DEFLATED'
+    elif compressed_size == uncompressed_size:
```

Comment:
You could have deflate-compressed data that simply happens to be the same size as the uncompressed data, resulting in a false positive here.

Reply:
Indeed, fair point! I am trying to create a "golden" rule here to be valid for the different Android versions and its tricky. There is still one edge case not properly covered, though I have not stumbled upon malware using it !yet!.

Reply:
> its tricky

I can imagine :)

> simply happens to be the same size as the uncompressed data

Or, being aware of tools like these, it might even be intentionally constructed as such :)